modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/cards/Cards.tsx

```diff
@@ -150,75 +141,72 @@ const Card = forwardRef<HTMLDivElement, any>(
-									executeAsyncItemAction,
-									highlightItems,
-									infoPanelOpen,
-									itemData: item,
-									itemId: selectedItemKey,
-									loadData,
-									onActionDropdownItemClick,
-									onInfoPanelToggleButtonClick,
-									onItemSelectionChange,
-									openModal,
-									openSidePanel,
-									toggleItemInlineEdit,
-								});
-							},
-						};
-					})}
-					description={localizedDescription}
-					href={(schema.link && item[schema.link]) || null}
-					imgProps={imageProps}
-					labels={getLabels(item)}
-					onClick={
-						selectable
-							? (event: any) => {
-									const target = getSelectionTrigger(event);
+		const props = {
+			actions: formattedActions?.map((action: IItemsActions) => ({
+				...action,
+				href: isLink(action.target, null)
+					? formatActionURL(action.href, item, action.target)
+					: null,
+				onClick: (event: Event) => {
+					handleActionClick({
+						action,
+						event,
+						executeAsyncItemAction,
+						highlightItems,
+						itemData: item,
+						itemId: selectedItemKey,
+						loadData,
+						onActionDropdownItemClick,
+						onInfoPanelToggleButtonClick,
+						openModal,
+						openSidePanel,
+						toggleItemInlineEdit,
+					});
+				},
+			})),
+			description: getLocalizedValue(item, schema.description)?.value,
+			href: (schema.link && item[schema.link]) || null,
+			imgProps:
+				schema.image &&
+				imagePropsTransformer(
+					getLocalizedValue(item, schema.image)?.value
+				),
+			labels: getLabels(item),
+			onClick: selectable
+				? (event: any) => {
+						const target = getSelectionTrigger(event);
 
-									if (target) {
-										onItemSelectionChange?.({
-											item,
-											trigger: target,
-										});
+						if (target) {
+							onItemSelectionChange?.({
+								item,
+								trigger: target,
+							});
 
-										event.preventDefault();
-									}
-								}
-							: undefined
-					}
-					onSelectChange={selectable ? () => undefined : undefined}
-					selectableType={
-						selectionType === 'single' ? 'radio' : 'checkbox'
-					}
-					selected={cardSelected}
-					stickerProps={
-						(schema.sticker && item[schema.sticker]) || null
+							event.preventDefault();
+						}
 					}
-					symbol={schema.symbol && item[schema.symbol]}
-					title={localizedTitle}
+				: undefined,
+			onSelectChange: selectable ? () => undefined : undefined,
+			selectableType: selectionType === 'single' ? 'radio' : 'checkbox',
+			selected:
+				selectable &&
+				!!selectedItemsValue?.find(
+					(element) =>
+						selectedItemsKey && element === item[selectedItemsKey]
+				),
+			stickerProps: (schema.sticker && item[schema.sticker]) || null,
+			symbol: schema.symbol && item[schema.symbol],
+			title: getLocalizedValue(item, schema.title)?.value || '',
+		};
+
+		return (
+			<div ref={ref}>
+				<ClayCardWithInfo
+					{...{
+						...props,
+						...(activeView.setItemComponentProps?.({item, props}) ??
```

Comment:
A side effect of this is the dependency we introduce between the `setItemComponentProps` implementations and the underlying component used to render an item (`<ClayCardWithInfo>` in this case).

This is not bad per se, but if we want to change card view to, say, `<BaseCard>` there is a need to review all implementations to adapt the transformed component props.

This affects dev exp a little bit, DXP guys using this API shall know about this dependency.

Reply:
Yeah, the dependency is intentional. In the future, if we would want to change to a different component, we would need to be careful. Options would be:
- Copy the existing snapshot of API from Clay component.
- The new component can be a wrapper around Clay component, allowing us to safely add new props. We can pass `otherProps` to Clay component, so there is no breaking change.

It would be a breaking change if we add a new component with different API.  But if we are careful, we can always add new API, never change old one. Something like:
```
if (newProp) {
   return <CustomComponent {...allProps} />
}
else {
    return <ClayComponent {...otherProps} />
}
```